core/core.go

```diff
@@ -123,133 +120,6 @@ type Mounts struct {
 	Ipns mount.Mount
 }
 
-type ConfigOption func(ctx context.Context) (*IpfsNode, error)
-
-func NewIPFSNode(ctx context.Context, option ConfigOption) (*IpfsNode, error) {
-	node, err := option(ctx)
-	if err != nil {
-		return nil, err
-	}
-
-	if node.ctx == nil {
-		node.ctx = ctx
-	}
-	if node.proc == nil {
-		node.proc = goprocessctx.WithContextAndTeardown(node.ctx, node.teardown)
-	}
-
-	success := false // flip to true after all sub-system inits succeed
-	defer func() {
-		if !success {
-			node.proc.Close()
```

Comment:
Oh, i had just copied it from the node builder, which had copied it from this function before this function changed to call `proc.Close()`. Should be fixed now with the latest change